external-import/flashpoint/src/flashpoint_connector/config_loader.py

```
@@ -0,0 +1,338 @@
+        """
+        return self.model_dump(mode="json", context={"mode": "pycti"})
+
+    @classmethod
+    def settings_customise_sources(
+        cls,
+        settings_cls: type[BaseSettings],
+        init_settings: PydanticBaseSettingsSource,
+        env_settings: PydanticBaseSettingsSource,
+        dotenv_settings: PydanticBaseSettingsSource,
+        file_secret_settings: PydanticBaseSettingsSource,
+    ) -> tuple[PydanticBaseSettingsSource, ...]:
+        """
+        Customise the sources of settings for the connector.
+
+        This method is called by the Pydantic BaseSettings class to determine the order of sources
+
+        The configuration come in this order either from:
+            1. YAML file
+            2. .env file
+            3. Environment variables
+            4. Default values
+        """
+        if Path(settings_cls.model_config["yaml_file"] or "").is_file():  # type: ignore
+            return (YamlConfigSettingsSource(settings_cls),)
+        if Path(settings_cls.model_config["env_file"] or "").is_file():  # type: ignore
+            return (dotenv_settings,)
+        return (env_settings,)
+
+    @model_validator(mode="before")
+    @classmethod
+    def migrate_deprecated_interval(cls, data: dict) -> dict:
+        """
+        Env var `FLASHPOINT_INTERVAL` is deprecated.
+        This is a workaround to keep the old config working while we migrate to `CONNECTOR_DURATION_PERIOD`.
+        """
+        connector_data: dict = data.get("connector", {})
+        flashpoint_data: dict = data.get("flashpoint", {})
+
+        if interval := flashpoint_data.pop("interval", None):
+            warnings.warn(
+                "Env var 'FLASHPOINT_INTERVAL' is deprecated. Use 'CONNECTOR_DURATION_PERIOD' instead."
+            )
+            if connector_data.get("duration_period"):
+                raise ValueError(
+                    "Cannot set both `FLASHPOINT_INTERVAL` and `CONNECTOR_DURATION_PERIOD`."
+                )
```

I don't know if we should raise an error here or just override the config (duration_period takes priority on interval) WDYT? Just wondering